packages/core/contracts/financial-templates/funding-rate-store/implementation/FundingRateStore.sol

```diff
@@ -79,14 +78,32 @@ contract FundingRateStore is FundingRateStoreInterface, Testable, Lockable {
     }
 
     struct FundingRateRecord {
-        FixedPoint.Signed rate; // Current funding rate.
-        uint256 proposeTime; // Time at which current funding rate was proposed.
+        // Current funding rate.
+        FixedPoint.Signed rate;
+        // Time at which current funding rate was proposed.
+        uint256 proposeTime;
         Proposal proposal;
-        FixedPoint.Unsigned rewardRatePerSecond; // Percentage of 1 E.g., .1 is 10%.
+    }
+
+    struct RecordParams {
+        // Liveness period for an update to value in RecordParams to become official.
+        uint256 paramUpdateLiveness;
+        // Reward rate paid to successful proposers. Percentage of 1 E.g., .1 is 10%.
+        FixedPoint.Unsigned rewardRatePerSecond;
+        // Bond % (of given contract's PfC) that must be staked by proposers. Percentage of 1, e.g. 0.0005 is 0.05%
+        FixedPoint.Unsigned proposerBondPct;
+    }
+
+    struct RecordParamsForContract {
+        // Upgradeable params specific to this record.
+        RecordParams current;
+        RecordParams pending;
+        uint256 pendingPassedTimestamp;
     }
 
     enum ProposalState { None, Pending, Expired }
 
+    mapping(address => RecordParamsForContract) public recordParams;
```

Comment:
can you add some more comments on these data structures, perhaps above them just to help with explaining exactly what they are used for like we have in other contracts?

Reply:
definitely

Reply:
done